tools/src/icon4pytools/py2fgen/wrappers/common.py

```diff
@@ -10,35 +10,39 @@
 import logging
 
 from icon4py.model.common import dimension as dims
-from icon4py.model.common.decomposition import definitions, mpi_decomposition as mpi
+from icon4py.model.common.decomposition import definitions
 from icon4py.model.common.grid import base, horizontal, icon
-from icon4py.model.common.settings import xp
+from icon4py.model.common.utils.gt4py_field_allocation import NDArray
 
+from icon4pytools.py2fgen.settings import config
+
+
+xp = config.array_ns
 
 log = logging.getLogger(__name__)
 
 
 def adjust_fortran_indices(inp: xp.ndarray, offset: int) -> xp.ndarray:
     """For some Fortran arrays we need to subtract 1 to be compatible with Python indexing."""
-    return xp.subtract(inp.ndarray, offset)
+    return xp.subtract(inp, offset)
 
 
 def construct_icon_grid(
-    cell_starts: xp.ndarray,
-    cell_ends: xp.ndarray,
-    vertex_starts: xp.ndarray,
-    vertex_ends: xp.ndarray,
-    edge_starts: xp.ndarray,
-    edge_ends: xp.ndarray,
-    c2e: xp.ndarray,
-    e2c: xp.ndarray,
-    c2e2c: xp.ndarray,
-    e2c2e: xp.ndarray,
-    e2v: xp.ndarray,
-    v2e: xp.ndarray,
-    v2c: xp.ndarray,
-    e2c2v: xp.ndarray,
-    c2v: xp.ndarray,
+    cell_starts: NDArray,
```

Comment:
you have the `xp` import as before in `wrappers.common`: 
You can use `xp.ndarray` as before, then you dont need to import another module (gt4py_field_allocation.py) from the model here, which is preferable.